R/plugin-providers.R

```diff
@@ -36,8 +38,18 @@ addProviderTiles <- function(
   provider,
   layerId = NULL,
   group = NULL,
-  options = providerTileOptions()
+  options = providerTileOptions(),
+  ...,
+  .check = TRUE
 ) {
+  if (.check) {
+    loaded_providers <- leaflet.providers::providers_loaded()
+    provider <- match.arg(
+      arg = provider,
+      choices = unlist(use.names = FALSE, loaded_providers$providers),
+      several.ok = FALSE
+    )
```

Comment:
Thank you for the contribution! Two concerns with this part:

1. `match.arg` will do partial matches, which feels undesirable here
2. The error message will be very long:
```
'provider' should be one of “OpenStreetMap”, “OpenStreetMap.Mapnik”, “OpenStreetMap.DE”, “OpenStreetMap.CH”, “OpenStreetMap.France”, “OpenStreetMap.HOT”, “OpenStreetMap.BZH”, “MapTilesAPI”, “MapTilesAPI.OSMEnglish”, “MapTilesAPI.OSMFrancais”, “MapTilesAPI.OSMEspagnol”, “OpenSeaMap”, “OPNVKarte”, “OpenTopoMap”, “OpenRailwayMap”, “OpenFireMap”, “SafeCast”, “Stadia”, “Stadia.AlidadeSmooth”, “Stadia.AlidadeSmoothDark”, “Stadia.OSMBright”, “Stadia.Outdoors”, “Stadia.StamenToner”, “Stadia.StamenTonerBackground”, “Stadia.StamenTonerLines”, “Stadia.StamenTonerLabels”, “Stadia.StamenTonerLite”, “Stadia.StamenWatercolor”, “Stadia.StamenTerrain”, “Stadia.StamenTerrainBackground”, “Stadia.StamenTerrainLabels”, “Stadia.StamenTerrainLines”, “Thunderforest”, “Thunderforest.OpenCycleMap”, “Thunderforest.Transport”, “Thunderforest.TransportDark”, “Thunder`
```

 and actually it gets cut off there even though there are dozens and dozens more provider names.

What would you think of this instead?

```suggestion
    if (!provider %in% names(loaded_providers$providers)) {
      stop("Unknown tile provider '", provider, "; either use a known provider or pass `.check = FALSE` to `addProviderTiles()`')
    }
```

Reply:
Hi Joe - that makes sense to me - didn't think about the partial matching in `match.arg()`! Committed this change.